cpp/cmake/bob.cmake

```diff
@@ -263,36 +258,36 @@ function(bob_begin_cxx_flags)
       set(FLAGS "${FLAGS} -g")
     endif()
     # -----------
-    bob_cxx_pedantic_flags()
+    if(${PROJECT_NAME}_CXX_WARNINGS)
+      bob_cxx_pedantic_flags(FLAGS)
+    endif()
     set(CMAKE_CXX_FLAGS "${FLAGS}" PARENT_SCOPE)
   endif()
 endfunction(bob_begin_cxx_flags)
 
 function(bob_cxx_standard_flags standard)
-  set(FLAGS "${CMAKE_CXX_FLAGS}")
-  set(FLAGS "${FLAGS} --std=${standard}")
-  if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  if(standard GREATER 98 AND CMAKE_CXX_COMPILER_ID MATCHES "Clang")
     if(${PROJECT_NAME}_CXX_WARNINGS)
       set(FLAGS "${FLAGS} -Wno-c++98-compat-pedantic -Wno-c++98-compat")
     endif()
   endif()
-  set(CMAKE_CXX_FLAGS "${FLAGS}" PARENT_SCOPE)
+  set(CMAKE_CXX_STANDARD ${standard})
```

Comment:
I'm just wondering… does `CMAKE_CXX_STANDARD_REQUIRED` need to be set, too?

Reply:
I didn't know about this variable. IMO, if you call `bob_cxx17_flags`, you expect C++17, so this is a requirement. I will update.

Reply:
There's more still, see i.e. here: https://crascit.com/2015/03/28/enabling-cxx11-in-cmake/

Reply:
842aff2 and 890a778 address this conversation